langchain4j-azure-ai-search/src/main/java/dev/langchain4j/rag/content/retriever/azure/search/AzureAiSearchFilterMapper.java

```diff
@@ -0,0 +1,7 @@
+package dev.langchain4j.rag.content.retriever.azure.search;
+
+import dev.langchain4j.store.embedding.filter.Filter;
+
+public interface AzureAiSearchFilterMapper {
```

Comment:
Is there a reason to have this interface and expose it to the user?
What are the scenarios when this can be useful?

Reply:
My idea is that the default mapper I wrote is based on the default structure of index created in azure ai search. I mean there is an attribute named "metadata" with an attribute named "attributes" that represent a map key/value.
But if someone define an other structure and want to filter then he'll have to adapt the format of filter string generated.

Reply:
AFAIU "metadata" and "attributes" are hardcoded inside the `AbstractAzureAiSearchEmbeddingStore` and cannot be changed, so at this point there is no way to set another names/structure.

This is not critical, but exposing this interface does not seem to have benefits, but it increases the overall complexity.